### PR TITLE
Adds coloured sprays for holoparasites.

### PR DESCRIPTION
**Pull request by ghost**
http://i.imgur.com/gpfGaqV.jpg
Adds colour respective sprays for all tech holoparasites.
Green is still kept as default.

### CODE REVIEW

**Comment by Spacedong** [comment]
no

**Comment by Voratius** [comment]
spacedong your opinion doesnt really matter on PRs.

anyways, yeah, what harm could it do? +1

**Comment by ghost** [comment]
yessssss

**Comment by dreamcrab** [comment]
yea this is cool +1738

**Comment by JohnGinnane** [comment]
Would it not make more sense to have one single projectile and had a colour variable? All of those projectile objects are exactly the same _except_ the colour

**Comment by dreamcrab** [comment]
i was gonna ask the same thing but you can spam the shit outta them, would it lag?

**Comment by ghost** [comment]
It'd take a bit of refactoring to make the spray a greyscale shot since shoot code in simple_animal hostile is pretty rigid.
If it would be preferable then I can try doing that again.

**Comment by Gandalf2k15** [comment]
Code is sound, however the thing John pointed out would make things neater, but if its too much it should be fine.

**Comment by Nebyab** [comment]
okay

**Comment by ghost** [comment]
I'm gonna try the grayscale colour thing again tomorrow, please don't merge until I've tried that.

**Comment by ghost** [comment]
untitled2
This is the current state of the new system.
The tech spray colours could use some sprucing up.
I'd also like a little bit of help to make my code less snowflakey.

File: code/modules/mob/living/simple_animal/guardian/guardian.dm [hunk 1]
```diff
@@ -407,22 +408,7 @@
...
-/obj/item/projectile/guardian/pink
-	icon_state = "guardianpink"
-
-/obj/item/projectile/guardian/orange
-	icon_state = "guardianorange"
-
-/obj/item/projectile/guardian/neon
-	icon_state = "guardianneon"
-
+	mouse_opacity = 1
```
Comment (ghost) [inline code/modules/mob/living/simple_animal/guardian/guardian.dm]: This is for testing purposes.

**Comment by ghost** [comment]
untitled2
And that is probably final for now.
The ghost looking ones use the built in HTML colours which is why they don't look at good as the holoparasites which are individually coloured.

**Comment by ghost** [comment]
This could probably reuse headmin revaluation since it changes it quite a bit.

**Comment by JohnGinnane** [comment]
Putting code review back on since you've revamped how the colours work.

File: code/modules/mob/living/simple_animal/guardian/guardian.dm [hunk 2]
```diff
@@ -490,6 +491,29 @@
 				if(G.summoner)
 					G.summoner << "<span class='danger'><B>[AM] has crossed your surveillance trap at [get_area(snare_loc)].</span></B>"
 
+/mob/living/simple_animal/hostile/guardian/ranged/Shoot(atom/targeted_atom)
+	var/obj/item/projectile/guardian/shard = ..()
+	if(!shard)
+		return
+	shard.color = guardiancolor
+
+/proc/techcolor2hex(var/colour)
```
Comment (Carbonhell) [inline code/modules/mob/living/simple_animal/guardian/guardian.dm:500]: Unneeded proc pretty sure(Check through,i'm not 100% sure), cause color can also be a word and not necessarily an hex code

**Comment by Carbonhell** [comment]
Fix the code i commented

File: code/modules/mob/living/simple_animal/guardian/guardian.dm (hunk 2 above)
Comment (JohnGinnane) [inline code/modules/mob/living/simple_animal/guardian/guardian.dm:500]: In this case colour is a word that's defined in guardian. It's what's passed in. This is similar to color2hex but while this is more snowflakey it means the colours match the guardians better, e.g. "red" would be 0xFF0000 while the red guardian might be more 0xFF2211

Comment (ghost) [inline code/modules/mob/living/simple_animal/guardian/guardian.dm:500]: Colours like neon aren't defined in color2hex.
The snowflake proc is to choose colors best suited for the guardians as John stated.

File: code/modules/mob/living/simple_animal/guardian/guardian.dm [hunk 3]
```diff
@@ -669,11 +694,11 @@
 			G.animated_manifest = TRUE
 			user << "[G.tech_fluff_string]."
 			G.speak_emote = list("states")
+			G.guardiancolor = techcolor2hex(colour)
 		if("bio")
 			user << "[G.bio_fluff_string]."
 			G.attacktext = "swarms"
 			G.speak_emote = list("chitters")
-	return G
```
Comment (JohnGinnane) [inline code/modules/mob/living/simple_animal/guardian/guardian.dm:676]: Why did you remove "return G"?

**Comment by JohnGinnane** [comment]
Code looks good, do we need head input?

**Comment by dreamcrab** [comment]
if shes working, go ahead and merge it up dog

**Comment by JohnGinnane** [comment]
Aw yiss
